packages/page-supersig/src/modals/IdentityMain.tsx

```diff
@@ -0,0 +1,269 @@
+// Copyright 2017-2022 @polkadot/app-accounts authors & contributors
+// SPDX-License-Identifier: Apache-2.0
+
+import type { Data, Option } from '@polkadot/types';
+import type { Registration } from '@polkadot/types/interfaces';
+
+import React, { useEffect, useState } from 'react';
+import styled from 'styled-components';
+
+import { Input, InputBalance, Modal, Toggle, TxButton } from '@polkadot/react-components';
+import { getAddressMeta } from '@polkadot/react-components/util';
+import { useApi, useCall } from '@polkadot/react-hooks';
+import { u8aToString } from '@polkadot/util';
+
+import { useTranslation } from '../translate';
+
+interface Props {
+  address: string;
+  className?: string;
+  onClose: () => void;
+}
+
+interface WrapProps {
+  children: React.ReactNode;
+  onChange: (isChecked: boolean) => void;
+  value: boolean;
+}
+
+interface ValueState {
+  info: Record<string, unknown>;
+  okAll: boolean;
+  okDisplay?: boolean;
+  okEmail?: boolean;
+  okLegal?: boolean;
+  okRiot?: boolean;
+  okTwitter?: boolean;
+  okWeb?: boolean;
+}
+
+const WHITESPACE = [' ', '\t'];
+
+function setData (data: Data, setActive: null | ((isActive: boolean) => void), setVal: (val: string) => void): void {
+  if (data.isRaw) {
+    setActive && setActive(true);
+    setVal(u8aToString(data.asRaw.toU8a(true)));
+  }
+}
+
+function WrapToggle ({ children, onChange, value }: WrapProps): React.ReactElement<WrapProps> {
+  const { t } = useTranslation();
+
+  return (
+    <div className='toggle-Wrap'>
+      {children}
+      <Toggle
+        isOverlay
+        label={t<string>('include field')}
+        onChange={onChange}
+        value={value}
+      />
+    </div>
+  );
+}
+
+function checkValue (hasValue: boolean, value: string | null | undefined, minLength: number, includes: string[], excludes: string[], starting: string[], notStarting: string[] = WHITESPACE, notEnding: string[] = WHITESPACE): boolean {
+  return !hasValue || (
+    !!value &&
+    (value.length >= minLength) &&
+    includes.reduce((hasIncludes: boolean, check) => hasIncludes && value.includes(check), true) &&
+    (!starting.length || starting.some((check) => value.startsWith(check))) &&
+    !excludes.some((check) => value.includes(check)) &&
+    !notStarting.some((check) => value.startsWith(check)) &&
+    !notEnding.some((check) => value.endsWith(check))
+  );
+}
+
+function IdentityMain ({ address, className = '', onClose }: Props): React.ReactElement<Props> {

```

Comment:
All this stuff seems just duplicated from accounts.

We want integrated pages, this is a stand-alone creation using the apps UI codebase.

We cannot have the same stuff in multiple places, this just means we double the workload. (This is already in need of an update since there is an open PR that extends this specifically)

Reply:
All the duplicates were just brought in to see how they work, they are not actually used. Will clean.